packages/date-picker/src/DatePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,247 @@
+  description,
+  onChange,
+  labelText,
+  isDisabled = false,
+  classNameOverride,
+  disabledDates,
+  disabledDaysOfWeek,
+  disabledRange,
+  disabledBeforeAfter,
+  disabledBefore,
+  disabledAfter,
+  firstDayOfWeek = 1,
+  initialMonth,
+  selectedDateRange,
+  value,
+  ...inputProps
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const [referenceElement, setReferenceElement] =
+    useState<HTMLDivElement | null>(null)
+  const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(
+    null
+  )
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    modifiers: [
+      {
+        name: "offset",
+        options: {
+          offset: [0, 15],
+        },
+      },
+    ],
+    placement: "bottom-start",
+  })
+
+  const handleOpenClose = () => {
+    setIsOpen(!isOpen)
+  }
+
+  const handleReturnFocus = () => {
+    if (buttonRef.current) {
+      buttonRef.current.focus()
+    }
+  }
+
+  const disabledDays = calculateDisabledDays(
+    disabledDates,
+    disabledDaysOfWeek,
+    disabledRange,
+    disabledBeforeAfter,
+    disabledBefore,
+    disabledAfter
+  )
+
+  const handleDayClick = (day: Date, modifiers: DayModifiers) => {
+    /** react-day-picker will fire events for disabled days by default.
+     *  We're checking here if it includes the CSS Modules class for disabled
+     *  on the modifier to then return early.
+     * */
+    if (Object.keys(modifiers).includes(defaultCalendarClasses.disabled)) {
+      return
+    }
+
+    if (!selectedDateRange) return
+
+    /** If user has already selected range and then selects again, treat first
+     * click as the start of the new range.
+     *  */
+    if (isSelectingFirstDay(selectedDateRange, day)) {
+      onChange({
+        from: day,
+        to: null,
+      })
+    } else {
+      // Otherwise, treat click as the final selection.
+      onChange({
+        from: selectedDateRange.from,
+        to: day,
+      })
+      handleOpenClose()
+    }
+  }
+
+  const isSelectingFirstDay = (range: RangeModifier, day: Date) => {
+    const isBeforeFirstDay =
+      !!range.from && DateUtils.isDayBefore(day, range.from)
+
+    const isRangeSelected = !!range.from && !!range.to
+
+    return !range.from || isBeforeFirstDay || isRangeSelected
+  }
+
+  const modifiers: RangeModifier = {
+    from: selectedDateRange?.from,
+    to: selectedDateRange?.to,
+  }
+
+  return (
+    <div ref={wrapperRef}>
+      <div ref={setReferenceElement}>
+        <Label disabled={isDisabled} htmlFor={id} labelText={labelText} />
+        <button
+          className={cx(
+            datePickerStyles.button,
+            datePickerStyles.withStartIconAdornment,
+            {
+              [datePickerStyles.disabled]: isDisabled,
+            }
+          )}
+          id={id}
+          disabled={isDisabled}
+          ref={buttonRef}
+          onClick={handleOpenClose}
```

Comment:
I was trying to get the screen reader to announce the selection after the picker closes but was able to focus on the date range within.

I had a look at the [W3 docs](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html) for date pickers and one of their suggested patterns is to add a aria-label to the button with the date to ensure that a screen reader announces the users selection